kernel/src/syscall/rt_sigaction.rs

```diff
@@ -77,18 +84,40 @@ pub fn sys_rt_sigaction(
 // pending and whose default action is to ignore the signal
 // (for example, SIGCHLD), shall cause the pending signal to
 // be discarded, whether or not it is blocked
-fn discard_signals_if_ignored(ctx: &Context, signum: SigNum, sig_action: &SigAction) {
-    if !sig_action.will_ignore(signum) {
+fn discard_signals_if_ignored(ctx: &Context, signum: SigNum) {
+    let mask = SigSet::new_full() - signum;
+
+    for task in ctx.process.tasks().lock().as_slice() {
+        let Some(posix_thread) = task.as_posix_thread() else {
+            continue;
+        };
+
+        while posix_thread.dequeue_signal(&mask).is_some() {}
+    }
```

Comment:
Should we deal with the process-wide sigqueue as well?

Reply:
The process-wide signal queue is already handled by `posix_thread.dequeue_signal`, which dequeues signals from both the thread's signal queue and the process-wide signal queue.